lute/cli/include/lute/compile.h

```diff
@@ -67,3 +67,27 @@ struct LuteDecodeResult
     size_t compressedPayloadSizeBytes = 0;
     size_t uncompressedPayloadSizeBytes = 0;
 };
+
+/**
+ * Manages creating and reading Lute executables with embedded bytecode bundles.
+ *
+ * Binary format (from end of file, reading backward):
+ *   [lute runtime executable's bytes]
+ *   [compressed bundle data]
+ *   [compressed_size: uint64_t]
+ *   [uncompressed_size: uint64_t]
+ *   [num_files: uint32_t]
+ *   [entry_point_path_string: char[entry_point_path_length]]
+ *   [entry_point_path_length: uint32_t]
+ *   [MAGIC_FLAG: "LUTEBYTE" (8 bytes)]
+ */
```

Comment:
Is there a good reason that this binary format is backwards? It really feels like it should just be `LUTEBYTE` followed by the rest of the data in order, and we should scan to the magic flag and then read forward from there. This is what I would _expect_, so I think there should be a clear reason why the format deviates from that.

Reply:
This actually shows a bug with my implementation - I was reading the file into memory before checking the LUTEBYTE at the end, when we can just seek to the end and check LUTEBYTE before reading this into memory..